### PR TITLE
[RU] Add Bypass lesson translation

### PR DESCRIPTION
**Pull request by brain-geek**

### CODE REVIEW

**Comment by brain-geek** [comment]
Hey @elixirschool/russian . Could you look at this? I believe there will be quite some comments.

File: ru/lessons/libraries/bypass.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
+---
+version: 1.0.0
+title: Bypass
+---
+
+При тестировании приложений часто идет речь о запросах к внешним сервисам.
+Мы даже можем захотеть симулировать различные ситуации типа неожиданных ошибок со стороны сервера.
+Реализация подобного подхода в Эликсире требует дополнительных инструментов.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Реализация подобного подхода в Elixir требует дополнительных инструментов.
```
IIRC we [didn't translate the language name](https://github.com/elixirschool/elixirschool/search?q=%D0%AD%D0%BB%D0%B8%D0%BA%D1%81%D0%B8%D1%80).

File: ru/lessons/libraries/bypass.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+В этом уроке мы рассмотрим как библиотека [bypass](https://github.com/PSPDFKit-labs/bypass) может помочь с таким тестированиием.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
В этом уроке мы рассмотрим как библиотека [bypass](https://github.com/PSPDFKit-labs/bypass) может помочь с таким тестированием.
```

File: ru/lessons/libraries/bypass.md [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+{% include toc.html %}
+
+## Что такое Bypass?
+
+[Bypass](https://github.com/PSPDFKit-labs/bypass) описывается как "быстрый способ создать заглушку которая может быть использована вместо реального HTTP сервера для возврата подготовленных ответов на запросы клиентов".
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
[Bypass](https://github.com/PSPDFKit-labs/bypass) описывается как "быстрый способ создать заглушку, которая может быть использована вместо реального HTTP-сервера для возврата подготовленных ответов на запросы клиентов".
```

File: ru/lessons/libraries/bypass.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+Что это значит?
+"Под капотом" Bypass - OTP приложение, которое имитирует внешний веб-сервер, слушающий и отвечающий на запросы.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Под капотом Bypass — OTP-приложение, которое имитирует внешний веб-сервер, слушающий и отвечающий на запросы.
```

All the previous occurrences of `под капотом` [were not enclosed in quotes](https://github.com/elixirschool/elixirschool/search?q=%D0%BF%D0%BE%D0%B4%20%D0%BA%D0%B0%D0%BF%D0%BE%D1%82%D0%BE%D0%BC).

File: ru/lessons/libraries/bypass.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+Используя предопределенные ответы внешних сервисов мы можем проверить любые возможные сценарии - как неожиданные отказы сервисов и ошибки, так и ожидаемые стандартные сценарии. И всё это - без выполнения внешних запросов
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Используя предопределённые ответы, мы можем проверить любые возможные сценарии — как неожиданные отказы сервисов и ошибки, так и ожидаемые стандартные сценарии. И всё это без выполнения внешних запросов
```

File: ru/lessons/libraries/bypass.md [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+
+## Использование Bypass
+
+Для того чтобы лучше продемонстрировать возможности Bypass мы построим простую утилиту для проверки работоспособности списка сайтов.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Чтобы лучше продемонстрировать возможности Bypass, мы напишем простую утилиту для проверки работоспособности списка сайтов.
```

File: ru/lessons/libraries/bypass.md [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+Для этого мы создадим новый проект с супервизором и GenServer-ом для проверки доменов по настраиваемому интервалу времени.
+Используя Bypass в тестах, мы сможем проверить что приложение корректно работает в разных ситуациях.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Используя Bypass в тестах, мы сможем проверить, что приложение корректно работает в разных ситуациях.
```

File: ru/lessons/libraries/bypass.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+
+_Замечание_: Если хотите посмотреть финальный код, его можно найти в репозитории Elixir School [Clinic](https://github.com/elixirschool/clinic).
+
+К этому моменту вы должны уметь создать новый Mix проект и добавить необходимые зависимости, потому сфокусируемся именно на тестируемых частях.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
К этому моменту вы должны уметь создавать новый Mix проект и добавлять необходимые зависимости, потому сфокусируемся именно на тестируемых частях.
```

File: ru/lessons/libraries/bypass.md [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+
+Если нужно вспомнить как это делается, Можно заглянуть в раздел [Создание проекта](/../../basics/mix/#Создание-проекта) соответствующего урока [Mix](/../../basics/mix/).
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Если нужно вспомнить, как это делается, можно заглянуть в раздел [Создание проекта](../../basics/mix/#Создание-проекта) соответствующего урока [Mix](../../basics/mix/).
```

File: ru/lessons/libraries/bypass.md [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+
+
+Давайте начнем с создания нового модуля, которые будет выполнять сами запросы к доменам.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Давайте начнём с создания нового модуля, который будет выполнять сами запросы к доменам.
```

File: ru/lessons/libraries/bypass.md [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+С использованием [HTTPoison](https://github.com/edgurgel/httpoison) мы создадим функцию `ping/1`, которая принимает URL и возвращает `{:ok, body}` для ответом с кодом 200 и `{:error, reason}` для остальных:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
С использованием [HTTPoison](https://github.com/edgurgel/httpoison) мы создадим функцию `ping/1`, которая принимает URL и возвращает `{:ok, body}` для ответов с кодом 200 и `{:error, reason}` для остальных:
```

File: ru/lessons/libraries/bypass.md [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+  defp response({:ok, %{status_code: status_code}}), do: {:error, "HTTP Status #{status_code}"}
+  defp response({:error, %{reason: reason}}), do: {:error, reason}
+end
+```
+
+Стоит заметить, что мы _не_ используем GenServer и для этого есть причина:
+Вынося функциональность из GenServer, мы имеем возможность протестировать наш код без лишней сложности.
+
+Код на месте, пора начать с тестами.
+Для начала работы с Bypass нужно удостовериться в том что он работает.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Для начала работы с Bypass нужно убедиться, что он работает.
```

File: ru/lessons/libraries/bypass.md [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+Для этого, приведем `test/test_helper.exs` к такому виду:
+
+```elixir
+ExUnit.start()
+Application.ensure_all_started(:bypass)
+```
+
+Теперь, когда понятно что Bypass работает в тестах, пора перейти к самому тесту `test/clinic/health_check_test.exs` и закончить настройку.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Теперь, когда понятно, что Bypass работает в тестах, пора перейти к самому тесту `test/clinic/health_check_test.exs` и закончить настройку.
```

File: ru/lessons/libraries/bypass.md [hunk 14]
```diff
@@ -0,0 +1,272 @@
...
+  setup do
+    bypass = Bypass.open()
+    {:ok, bypass: bypass}
+  end
+end
+```
+
+Пока мы будем использовать Bypass по его порту по умолчанию, но если нужно будет его изменить (что мы сделаем позднее), можно передать параметр `:port` функции `Bypass.open/1`. Например, `Bypass.open(port: 1337)`.
+Теперь мы готовы работать с Bypass.
+Начнем с проверки позитивного сценария:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Начнём с проверки успешного сценария:
```

File: ru/lessons/libraries/bypass.md [hunk 15]
```diff
@@ -0,0 +1,272 @@
...
+    Bypass.expect(bypass, fn conn ->
+      Plug.Conn.resp(conn, 200, "pong")
+    end)
+
+    assert {:ok, "pong"} = HealthCheck.ping("http://localhost:#{bypass.port}")
+  end
+end
+```
+
+Тест прост и если мы его запустим - он пройдет. Но давайте посмотрим что делает каждая часть кода.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Тест прост, и если мы его запустим, он пройдет. Но давайте посмотрим, что делает каждая часть кода.
```

File: ru/lessons/libraries/bypass.md [hunk 16]
```diff
@@ -0,0 +1,272 @@
...
+Первое же что мы видим в тесте - вызов функции `Bypass.expect/2`:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Первое же, что мы видим в тесте — вызов функции `Bypass.expect/2`:
```

File: ru/lessons/libraries/bypass.md [hunk 17]
```diff
@@ -0,0 +1,272 @@
...
+
+```elixir
+Bypass.expect(bypass, fn conn ->
+  Plug.Conn.resp(conn, 200, "pong")
+end)
+```
+
+`Bypass.expect/2` принимает подключение к Bypass и функцию с одним аргументом, которая изменит подключение и вернет его. Также это позволяет делать проверки на запросе для проверки тот ли он, что мы ожидали. Давайте обновим тест для проверки что это GET запрос к пути `/ping`:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
`Bypass.expect/2` принимает подключение к Bypass и функцию с одним аргументом, которая изменит подключение и вернёт его. Также это позволяет проверять, что это тот запрос, который мы ожидали. Давайте обновим тест для проверки что это GET-запрос к пути `/ping`:
```

File: ru/lessons/libraries/bypass.md [hunk 18]
```diff
@@ -0,0 +1,272 @@
...
+  end)
+
+  assert {:ok, "pong"} = HealthCheck.ping("http://localhost:#{bypass.port}/ping")
+end
+```
+
+В последней части теста мы используем `HealthCheck.ping/1` и проверяем что ответ правилен, как и ожидалось. Но что такое `bypass.port`?
+Bypass слушает на локальном порту и принимает эти запросы. И мы используем `bypass.port` для получения номера порта по умолчанию, так как мы не определили его самостоятельно в `Bypass.open/1`.
+
+Следующая часть - добавить тестовые сценарии для ошибок.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Следующая часть — добавить тестовые сценарии для ошибок.
```

File: ru/lessons/libraries/bypass.md [hunk 19]
```diff
@@ -0,0 +1,272 @@
...
+Мы можем начать с тестом, похожим на наш первый с небольшими изменениями: вернуть 500 код ответа и проверить что `{:error, reason}` был возвращен:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Мы можем начать с теста, похожего на наш первый с небольшими изменениями: вернуть 500 код ответа и проверить, что вернулся кортеж `{:error, reason}`:
```

File: ru/lessons/libraries/bypass.md [hunk 20]
```diff
@@ -0,0 +1,272 @@
...
+  Bypass.expect(bypass, fn conn ->
+    Plug.Conn.resp(conn, 500, "Server Error")
+  end)
+
+  assert {:error, "HTTP Status 500"} = HealthCheck.ping("http://localhost:#{bypass.port}")
+end
+```
+
+В этом сценарии нет ничего особенного, так что пора перейти к следующему: неожиданный отказ сервера.
+Для этого, мы не будем использовать `Bypass.expect/2` - вместо этого, `Bypass.down/1` будет полезен для сброса соединения:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Для этого мы не будем использовать `Bypass.expect/2` — вместо этого `Bypass.down/1` будет полезен для сброса соединения:
```

File: ru/lessons/libraries/bypass.md [hunk 21]
```diff
@@ -0,0 +1,272 @@
...
+
+```elixir
+test "request with unexpected outage", %{bypass: bypass} do
+  Bypass.down(bypass)
+
+  assert {:error, :econnrefused} = HealthCheck.ping("http://localhost:#{bypass.port}")
+end
+```
+
+Если мы запустим новые тесты, то все пройдет - как и ожидалось!
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Если мы запустим новые тесты, они все пройдут успешно, как и ожидалось!
```

File: ru/lessons/libraries/bypass.md [hunk 22]
```diff
@@ -0,0 +1,272 @@
...
+Теперь, когда модуль `HealthCheck` протестирован, мы можем перейти к следующей части - тестировать его вместе с планировщиком, основанным на GenServer.
+
+## Несколько внешних сайтов
+
+Для этого проекта, мы остановимся на минимальном планировщике и будем использовать `Process.send_after/3` для запуска повторяющихся проверок.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Для этого проекта мы остановимся на минимальном планировщике и будем использовать `Process.send_after/3` для запуска повторяющихся проверок.
```

File: ru/lessons/libraries/bypass.md [hunk 23]
```diff
@@ -0,0 +1,272 @@
...
+
+  Process.send_after(self(), :check, interval)
+
+  {:ok, {health_check, sites}}
+end
+```
+
+Теперь нужно определить функцию `handle_info/2` для получения сообщения `:check` отправленного с помощью `send_after/2`.
+Для упрощения всего мы будем передавать список сайтов в функцию `HealthCheck.ping/1` и выводить результат либо с помощью `Logger.info`, либо `Logger.error`.
+Код будет написан с учетом того что позже его можно будет улучшить.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Код будет написан с учетом того, что позже его можно будет улучшить.
```

File: ru/lessons/libraries/bypass.md [hunk 24]
```diff
@@ -0,0 +1,272 @@
...
+
+defp report({:ok, body}), do: Logger.info(body)
+defp report({:error, reason}) do
+  reason
+  |> to_string()
+  |> Logger.error()
+end
+```
+
+Как уже обсуждалось, мы передаем сайты в `HealthCheck.ping/1` и итерируем по результатам с помощью `Enum.each/2`, вызывая функцию `report/1` для каждого.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Как уже обсуждалось, мы передаём сайты в `HealthCheck.ping/1` и проходим по результатам с помощью `Enum.each/2`, вызывая функцию `report/1` для каждого.
```

File: ru/lessons/libraries/bypass.md [hunk 25]
```diff
@@ -0,0 +1,272 @@
...
+        :timer.sleep(10)
+      end)
+
+    assert output =~ "pong"
+    assert output =~ "HTTP Status 404"
+  end
+end
+```
+
+Мы полагаемся на тест реализации с `TestCheck` одновременно с `CaptureLog.capture_log/1` для проверки что соответствующие методы логированы.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Мы полагаемся на тест реализации с `TestCheck` одновременно с `CaptureLog.capture_log/1` для проверки, что соответствующие методы логированы.
```

File: ru/lessons/libraries/bypass.md [hunk 26]
```diff
@@ -0,0 +1,272 @@
...
+
+Теперь у нас есть работающие модули `Scheduler` и `HealthCheck`, давайте напишем интеграционный тест для проверки что все работает вместе. Нам понадобится Bypass для этого теста и нам прийдется работать с несколькими Bypass запросами в рамках одного теста, давайте посмотрим как это делать.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Теперь у нас есть работающие модули `Scheduler` и `HealthCheck`. Давайте напишем интеграционный тест для проверки, что всё работает вместе. Нам понадобится Bypass для этого теста и нам придётся работать с несколькими Bypass запросами в рамках одного теста. Давайте посмотрим, как это сделать.
```

File: ru/lessons/libraries/bypass.md [hunk 27]
```diff
@@ -0,0 +1,272 @@
...
+
+Помните `bypass.port` из более ранней части? Когда нам нужно работать с несколькими сайтами, опция `:port` незаменима. Как можно было уже догадаться, можно создать несколько Bypass подключения - каждое со своим портом, которые будут имитировать разные сайты. Мы начнем с изучения обновленного `test/clinic_test.exs`:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Помните `bypass.port` из более ранней части? Когда нам нужно работать с несколькими сайтами, опция `:port` незаменима. Как можно было уже догадаться, можно создать несколько Bypass-подключений — каждое со своим портом, которые будут имитировать разные сайты. Мы начнем с изучения обновленного `test/clinic_test.exs`:
```

File: ru/lessons/libraries/bypass.md [hunk 28]
```diff
@@ -0,0 +1,272 @@
...
+        :timer.sleep(10)
+      end)
+
+    assert output =~ "[info]  pong"
+    assert output =~ "[error] HTTP Status 500"
+  end
+end
+```
+
+В этом тесте не должно быть ничего неожиданного. Вместо создания одного `Bypass` подключения в `setup`, мы создаем два с передачей им портов 1234 и 1337. Дальше - вызовы `Bypass.expect/2` и тот же код что был у нас в `SchedulerTest` для запуска планировщика и проверки что соответствующие сообщения записываются в лог.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
В этом тесте не должно быть ничего неожиданного. Вместо создания одного `Bypass`-подключения в `setup`, мы создаем два с передачей им портов 1234 и 1337. Дальше — вызовы `Bypass.expect/2` и тот же код, что был у нас в `SchedulerTest` для запуска планировщика и проверки, что соответствующие сообщения записываются в лог.
```

File: ru/lessons/libraries/bypass.md [hunk 29]
```diff
@@ -0,0 +1,272 @@
...
+
+Вот и всё! Мы построили небольшую утилиту для информирования о проблемах с доменами и изучили как использовать Bypass для написания тестов работы с внешними сервисами.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Вот и всё! Мы написали небольшую утилиту для информирования о проблемах с доменами и изучили, как использовать Bypass для написания тестов работы с внешними сервисами.
```

**Comment by brain-geek** [comment]
@nscyclone Thanks for this huge amount of feedback! Applied. Also, I've re-read it and applied some stylistical changes as well. Could you proofread once again?

File: ru/lessons/libraries/bypass.md [hunk 30]
```diff
@@ -0,0 +1,271 @@
...
+```elixir
+test "request with unexpected outage", %{bypass: bypass} do
+  Bypass.down(bypass)
+
+  assert {:error, :econnrefused} = HealthCheck.ping("http://localhost:#{bypass.port}")
+end
+```
+
+Если мы запустим новые тесты, они все пройдут успешно, как и ожидалось!
+Теперь, когда модуль `HealthCheck` протестирован, мы можем перейти к следующей части - протестировать его вместе с планировщиком, основанным на GenServer.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Теперь, когда модуль `HealthCheck` протестирован, мы можем перейти к следующей части — протестировать его вместе с планировщиком, основанным на GenServer.
```

File: ru/lessons/libraries/bypass.md [hunk 31]
```diff
@@ -0,0 +1,271 @@
...
+    assert "GET" == conn.method
+    assert "/ping" == conn.request_path
+    Plug.Conn.resp(conn, 200, "pong")
+  end)
+
+  assert {:ok, "pong"} = HealthCheck.ping("http://localhost:#{bypass.port}/ping")
+end
+```
+
+В последней части теста мы используем `HealthCheck.ping/1` и проверяем что ответ правилен, как и ожидалось. Но что такое `bypass.port`?
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
В последней части теста мы используем `HealthCheck.ping/1` и проверяем, что ответ правилен, как и ожидалось. Но что такое `bypass.port`?
```

File: ru/lessons/libraries/bypass.md [hunk 32]
```diff
@@ -0,0 +1,271 @@
...
+Тест прост, и если мы его запустим, он пройдет. Но давайте посмотрим, что делает каждая часть кода.
+Первое же, что мы видим в тесте — вызов функции `Bypass.expect/2`:
+
+```elixir
+Bypass.expect(bypass, fn conn ->
+  Plug.Conn.resp(conn, 200, "pong")
+end)
+```
+
+`Bypass.expect/2` принимает подключение к Bypass и функцию с одним аргументом, которая изменит подключение и вернёт его. Также это позволяет проверять, что это тот запрос, который мы ожидали. Давайте обновим тест для проверки что это GET-запрос к пути `/ping`:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
`Bypass.expect/2` принимает подключение к Bypass и функцию с одним аргументом, которая изменит подключение и вернёт его. Также это позволяет проверять, что это тот запрос, который мы ожидали. Давайте обновим тест для проверки, что это GET-запрос к пути `/ping`:
```

File: ru/lessons/libraries/bypass.md [hunk 33]
```diff
@@ -0,0 +1,271 @@
+---
+version: 1.0.0
+title: Bypass
+---
+
+При тестировании приложений часто идет речь о запросах к внешним сервисам.
+Мы даже можем захотеть имитировать различные ситуации таких, как ошибки со стороны сервера.
+Реализация подобного подхода в Elixir требует дополнительных инструментов.
+
+В этом уроке мы рассмотрим как библиотека [bypass](https://github.com/PSPDFKit-labs/bypass) может помочь с таким тестированием.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
В этом уроке мы рассмотрим, как библиотека [bypass](https://github.com/PSPDFKit-labs/bypass) может помочь с таким тестированием.
```

File: ru/lessons/libraries/bypass.md [hunk 34]
```diff
@@ -0,0 +1,271 @@
+---
+version: 1.0.0
+title: Bypass
+---
+
+При тестировании приложений часто идет речь о запросах к внешним сервисам.
+Мы даже можем захотеть имитировать различные ситуации таких, как ошибки со стороны сервера.
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Мы даже можем захотеть имитировать такие различные ситуации, как ошибки со стороны сервера.
```

File: ru/lessons/libraries/bypass.md [hunk 35]
```diff
@@ -0,0 +1,271 @@
...
+Используя Bypass в тестах, мы сможем проверить, что приложение корректно работает в разных ситуациях.
+
+_Замечание_: Если хотите посмотреть финальный код, его можно найти в репозитории Elixir School [Clinic](https://github.com/elixirschool/clinic).
+
+К этому моменту вы должны уметь создавать новый Mix проект и добавлять необходимые зависимости, потому сфокусируемся именно на тестируемых частях.
+
+Если нужно вспомнить, как это делается, можно заглянуть в раздел [Создание проекта](../../basics/mix/#Создание-проекта) соответствующего урока [Mix](../../basics/mix/).
+
+Давайте начнём с создания нового модуля, который будет выполнять сами запросы к доменам.
+С использованием [HTTPoison](https://github.com/edgurgel/httpoison) мы создадим функцию `ping/1`, которая принимает URL и возвращает `{:ok, body}` для ответов с кодом 200 и `{:error, reason}` для остальных:
```
Comment (nscyclone) [inline ru/lessons/libraries/bypass.md]:
```suggestion
Используя [HTTPoison](https://github.com/edgurgel/httpoison), мы создадим функцию `ping/1`, которая принимает URL и возвращает `{:ok, body}` для ответов с кодом 200 и `{:error, reason}` для остальных:
```

**Comment by brain-geek** [comment]
@nscyclone Done. Damn, that was a lot of changes, Thanks!

**Pull request merged**